### PR TITLE
Jeremy/eng 744 contact host modal

### PR DESCRIPTION
**Pull request by jerminatorhits**
## Description
Why did you write this code?
When user clicks 'contact host' button, a modal should appear with a small form instead of opening the computer's native mailer

requires https://github.com/thebeetoken/beenest-backend/pull/643

## How to Test
- [ ] Visit XXX
- [ ] Click Y

Which devices did you test on?
- [ ] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?

## Learnings
Did you learn anything here you want to share with the team?

### CODE REVIEW

File: src/components/shared/ContactHostForm/ContactHostForm.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+              <Button
+                disabled={isSubmitting}
+                type="submit">
+                Send Message
+              </Button>
+            </Form>
+          )}
+        </Formik>
+        <div className="close" onClick={onClose}>
```
Comment (tc) [inline src/components/shared/ContactHostForm/ContactHostForm.tsx]: Dont we have a shared `close` component?

File: src/networking/bookings.ts [hunk 2]
```diff
@@ -292,6 +292,11 @@ export const GET_GUEST_SORTED_BOOKINGS = gql`
   fragment baseTripFields on Booking {
     checkInDate
     checkOutDate
+    host {
+      id
+      firstName
+      supportEmail
```
Comment (tc) [inline src/networking/bookings.ts]: do we still need the support email if this feature lets the guest can contact the host via a form?

Comment (jerminatorhits) [reply src/networking/bookings.ts]: ya i think we can deprecate `supportEmail`

File: src/components/shared/ContactHostForm/ContactHostForm.tsx (hunk 1 above)
Comment (jerminatorhits) [reply src/components/shared/ContactHostForm/ContactHostForm.tsx]: ya sry. newest code push should have <Closebutton />

**Comment by tc** [comment]
i clicked Contact Host on /listings/95-vic and got an email not the modal.

**Comment by jerminatorhits** [comment]
ahhh will need to replace in all `contact host` througout app

**Comment by tc** [comment]
There seems to be a huge `padding-bottom: 72px`; is this necessary or part of the design? Seems ok without it.

File: src/components/routes/Listing/Listing/ListingInformation.tsx [hunk 3]
```diff
@@ -120,11 +122,30 @@ const ListingInformation = ({ listing, host }: Props) => {
               // TODO: When page is refreshed, supportEmail is null due to firebase user query and our listing query running in parallel.
```
Comment (tc) [inline src/components/routes/Listing/Listing/ListingInformation.tsx]: can we remove this comment now that we don't use supportEmail?

File: src/components/shared/ContactHostForm/ContactHostForm.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+              bookingId,
+              listingId,
+              recipientId: host.id,
+            };
+            return contactUser(input)
+              .then((response: any) => {
+                this.setState({ response: response.data.contactUser });
+              })
+              .catch((error: Error) => {
+                alert(`${error}. If this continues to occur, please contact us at support@beetoken.com`);
```
Comment (tc) [inline src/components/shared/ContactHostForm/ContactHostForm.tsx]: use support@beenest

File: src/networking/bookings.ts (hunk 2 above)
Comment (tc) [reply src/networking/bookings.ts]: bump

**Comment by jerminatorhits** [comment]
good catch - padding removed. I think I initially had the padding there to sort of "hard code" the vertical spacing for desktop. Used Flex divs on top and bottom to evenly space the form vertically @kcvan hope this flies with u

File: src/components/routes/Listing/Listing/ListingInformation.tsx (hunk 3 above)
Comment (jerminatorhits) [reply src/components/routes/Listing/Listing/ListingInformation.tsx]: removed

File: src/components/shared/ContactHostForm/ContactHostForm.tsx (hunk 4 above)
Comment (jerminatorhits) [reply src/components/shared/ContactHostForm/ContactHostForm.tsx]: changed

**Comment by tc** [comment]
@kcvan was @jerminatorhits 's css changes fine?

File: src/components/shared/AlertCard/AlertCard.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  message: string;
+  onClose: () => void;
+  src: string;
+  srcColor: string;
+  title: string;
+}>
+
+const AlertCard = ({ buttonBackgroundColor, cta, ctaColor, message, onClose, src, srcColor, title }: Props) => {
+  return (
+    <Card
```
Comment (kcvan) [inline src/components/shared/AlertCard/AlertCard.tsx]: I think we can get rid of the height/width/padding properties. We can move the `AlertCardContainer` on the top level and then target the `Card` class name directly and add the height/width/padding there

Comment (kcvan) [reply src/components/shared/AlertCard/AlertCard.tsx]: will also need height and width 100% on the card itself

Comment (kcvan) [reply src/components/shared/AlertCard/AlertCard.tsx]: you can also have a default padding and overwrite it on the class

**Comment by kcvan** [comment]
LGTM besides minor changes i mentioned

File: src/components/shared/AlertCard/AlertCard.tsx (hunk 5 above)
Comment (jerminatorhits) [reply src/components/shared/AlertCard/AlertCard.tsx]: updated

File: src/networking/bookings.ts (hunk 2 above)
Comment (jerminatorhits) [reply src/networking/bookings.ts]: deleted all `supportEmail` declarations -- they were all in gql `networking` files
